java/ql/src/semmle/code/java/dataflow/FlowSources.qll

```diff
@@ -318,3 +319,24 @@ class AndroidIntentInput extends DataFlow::Node {
     )
   }
 }
+
+/** Exported Android `Intent` that may have come from a hostile application. */
+class ExportedAndroidIntentInput extends RemoteFlowSource {
+  ExportedAndroidIntentInput() {
+    exists(ExportableAndroidComponent exportedType | exportedType.isExported() |
+      exists(MethodAccess ma, AndroidGetIntentMethod m |
+        ma.getMethod().overrides*(m) and
+        this.asExpr() = ma and
+        exportedType = ma.getEnclosingCallable().getDeclaringType()
```

Comment:
```suggestion
        exportedType = ma.getMethod().getDeclaringType()
```
(As @aschackmull suggested)

Reply:
Sorry @aschackmull and @smowton, <code>ma.getMethod().getDeclaringType()</code> doesn't seem to help here since it always returns a type <code>android.app.Activity</code> instead of the real activity class <code>com.example.app.UnsafeActivity1</code> that we want to check whether it's exported or not.

How could we model this? Thanks.

Reply:
I conferred with @aschackmull -- `getReceiverType()` is actually the one to go for.

Reply:
Thanks @smowton and @aschackmull. I've updated the query.